### PR TITLE
Update formatting for references with delta percentage

### PR DESCRIPTION
**Pull request by RobinPapke**
* remove prefix and suffix for references with delta percent and set suffix to percentage sign
* check if value is integer first in formatting to prevent adding digits after the decimal point for set precisions

### CODE REVIEW

**Comment by mikeengland** [comment]
Out of interest, are there no tests for this area of the code?

**Comment by RobinPapke** [comment]
Seems I missed a test, gonna fix that now.

**Comment by x8lucas8x** [comment]
> Seems I missed a test, gonna fix that now.

I still see no test in the code after the update.

**Comment by twheys** [comment]
looks good but this needs to be fixed for all widgets I think, can you please check?

File: fireant/slicer/widgets/reacttable.py [hunk 1]
```diff
@@ -83,8 +83,13 @@ def __init__(self, item, reference):
             self.key = reference_key(item, reference)
             self.label = reference_label(item, reference)
 
-        self.prefix = item.prefix
-        self.suffix = item.suffix
+        if reference is not None and reference.delta_percent:
```
Comment (twheys) [inline fireant/slicer/widgets/reacttable.py]: Maybe you can move this logic into fireant.slicer.references so i can be reused across other widgets

```
def reference_prefix(metric, reference):
    """
    Return the prefix for a metric displayed for a reference (or no Reference)

    :return:
        A string that is used as the prefix for a reference metric.
    """
    if reference is not None and reference.delta_percent:
        return None
    return metric.prefix
```

Comment (twheys) [reply fireant/slicer/widgets/reacttable.py]: same for suffix

File: fireant/tests/slicer/widgets/test_reacttable.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from unittest import TestCase
 
 from fireant.slicer.totals import MAX_STRING
-from fireant.slicer.widgets.reacttable import ReactTable
+from fireant.slicer.widgets.reacttable import ReactTable, ReferenceItem
```
Comment (twheys) [inline fireant/tests/slicer/widgets/test_reacttable.py]: can you please fix your pycharm formatted to enable multiple imports to be chopped down onto multiple lines? this lowers the risk of running into merge conflicts.

**Comment by twheys** [comment]
still missing formatting for pandas widget

**Comment by RobinPapke** [comment]
Fixing tests

**Pull request merged**